packages/type-params/src/decode/index.ts

```diff
@@ -34,6 +35,10 @@ function decodeArray ([ type ]: Param$Type$Array, input: Uint8Array, version: En
   for (let index = 0; index < arrayLength; index++) {
     const decoded = decode(type, input.subarray(length), version);
 
+    if (type === 'AccountId') {
```

Comment:
I cannot be here. All type decoding is done is `decodeValue`. As it stands we now have a situation where exactly the same logic exists in 2 places with no benefit. 

Array will always call through to the base and the base will always end up in decodeValue.